packages/medusa/src/migrations/1698160215000-cart-sales-channels-link.ts

```diff
@@ -0,0 +1,34 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+import IsolateSalesChannelDomainFeatureFlag from "../loaders/feature-flags/isolate-sales-channel-domain"
+
+export const featureFlag = IsolateSalesChannelDomainFeatureFlag.key
+
+export class CartSalesChannelsLink1698160215000 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+        CREATE TABLE IF NOT EXISTS "cart_sales_channel"
+        (
+            "id"                character varying        NOT NULL,
+            "cart_id"           character varying        NOT NULL,
+            "sales_channel_id"  character varying,
+            "created_at"        TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+            "updated_at"        TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+            "deleted_at"        TIMESTAMP WITH TIME ZONE,
+            CONSTRAINT "cart_sales_channel_pk"              PRIMARY KEY ("id"),
+            CONSTRAINT "cart_sales_channel_cart_id_unique"  UNIQUE ("cart_id")
+            );
+
+        insert into "cart_sales_channel" (id, cart_id, sales_channel_id)
```

Comment:
self-review: how do we "sync" data when FF is turned on/off? With relations that had pivot tables before the link is introduced, it is straightforward but here we are introducing a pivot table; before we had OneToMany and SC id was just a column on the cart table

Reply:
How we did it with prices is that we created a migration PR in the core first to the exact relations and properties that we needed and then created the module